src/module/mm-util/helpers.ts

```diff
@@ -79,85 +79,75 @@ export class FetcherCache<A, T> {
   }
 }
 
-export async function mm_wrap_item<T extends EntryType & LancerItemType>(
-  item: LancerItem<T>,
+export async function mm_wrap_item<T extends LancerItemType>(
+  item: LancerItem,
   use_existing_ctx: OpCtx
 ): Promise<LiveEntryTypes<T>> {
   // Get the reg that'd hold this item
   let reg: FoundryReg;
 
-  // @ts-ignore 0.8
   if (item.parent != null) {
     // We are an owned item
-    // @ts-ignore 0.8
-    if(item.pack != null) {
+    if (item.pack != null) {
       // A compendium actor owned item
       reg = new FoundryReg({
         src: "comp_actor",
-        // @ts-ignore 0.8
         comp_id: item.pack,
-        // @ts-ignore 0.8
-        actor_id: item.actor.id,
+        actor_id: item.actor!.id!,
       });
-      //@ts-ignore 0.8
-    } else if(item.parent.isToken) {
+    } else if (item.parent.isToken) {
       // A token actor owned item
       reg = new FoundryReg({
         src: "scene_token",
-        // @ts-ignore 0.8
-        scene_id: item.actor.token.parent.id,
-        // @ts-ignore 0.8
-        token_id: item.actor.token.id
+        scene_id: item.actor!.token!.parent!.id!,
+        token_id: item.actor!.token!.id!,
       });
     } else {
       // A world actor owned item
       reg = new FoundryReg({
         src: "game_actor",
-        // @ts-ignore 0.8
-        actor_id: item.actor.id,
+        actor_id: item.actor!.id!,
       });
     }
   } else {
     // We are an unowned item
-    // @ts-ignore 0.8
-    if(item.pack != null) {
+    if (item.pack != null) {
       // An unowned compendium item
-      // @ts-ignore 0.8
-      if(is_core_pack_name(item.pack)) {
+      if (is_core_pack_name(item.pack)) {
         // An unowned core compendium item
         reg = new FoundryReg({
-          src: "comp_core"
+          src: "comp_core",
         });
       } else {
         // An unowned custom compendium item
         reg = new FoundryReg({
           src: "comp",
-          // @ts-ignore 0.8
-          comp_id: item.pack
+          comp_id: item.pack,
         });
       }
     } else {
       // An unowned game item
       reg = new FoundryReg({
-        src: "game"
+        src: "game",
       });
     }
   }
-  
+
   let ctx = use_existing_ctx || new OpCtx();
 
   // Load up the item. This _should_ always work
   // let ent = (await reg.get_cat(item.type).get_live(ctx, item.id)) as LiveEntryTypes<T>;
-  let cat = reg.get_cat(item.data.type) as FoundryRegCat<T>;
-  let ent = (await cat.dangerous_wrap_doc(ctx, item as any)) as LiveEntryTypes<T>; // Poor typescript doesn't know how to handle these
+  let cat = reg.get_cat(item.data.type);
+  // @ts-ignore dangerous_wrap_doc doesn't seem to exist? TODO
+  let ent = await cat.dangerous_wrap_doc(ctx, item);
```

Comment:
The only input I have is that I think this is a machine-mind bug, but I can't figure out what the code is doing well enough to give a decent bug report. My guess is that machine-mind's `Registry` should be generic and pass type data to `get_cat`. But it's a just a guess.

Reply:
`Registry` would be generic except that it's... very difficult to have a generic generic? Because it can't just be `Registry<CT extends RegCat<any>`, because then there's no way to have a `CT<Mech>` for instance - the wrapped genericness is lost. Trust me that this frustrates me as much as you and is something that would be solved by a proper macro system in TS, but alas. We spent literal hours discussing ways to make this work in the dev channel months ago, but ultimately decided it wasn't worth further attention.

As far as a solution: If you cast `cat` to a `FoundryRegCat<T>` it will have the functions described. Didn't realize that had just been TS-ignored, it should've been fairly simple

Reply:
That cast is pretty rough. It has to go through unknown. What about wrapping get_cat in the FoundryRegistry implementation? Then at least the casting is only in one place?

Reply:
oh yeah, I guess that'd work. If you could do that in my absence that would be appreciated (it should literally just be tighter signature + a super. call).

Reply:
Just pushed it.